09-SyOps.Rmd

```diff
@@ -1,58 +1,52 @@
+* report any security incidents, including a loss of data, in line with the relevant MoJ, HMPPS or HMCTS procedures;
+* report any breach of this acceptable use policy to the [Analytical Platform team](mailto:analytical-platform@digital.justice.gov.uk);
+* follow all relevant information governance procedures;
+* protect their login credentials appropriately;
+* create secure passwords following best practice guidelines (see [here](https://www.ncsc.gov.uk/blog-post/three-random-words-or-thinkrandom-0));
+* ensure that two-factor authentication is enabled when accessing GitHub and the Analytical Platform;
+* sign out of the Analytical Platform when access is not required;
+* understand they and MoJ have a legal responsibility to protect personal and sensitive information;
+* understand that their use of the Analytical Platform may be monitored;
+* ensure that all transfers of data onto and within the Analytical Platform are conducted safely and securely;
+* not access the Analytical Platform from any non-MoJ IT system, such as a personal computer;
+* not share their account or login credentials with any other person;
+* not use the same login credentials for more than one system or purpose;
+* not store any data on the Analytical Platform that is classified as SECRET or TOP SECRET;
+* not move any data to the Analytical Platform without completing a data movement form;
+* not attempt to access any data, apps or software on the Analytical Platform without the appropriate permission; and
+* not use the Analytical Platform to undertake any illegal activity or any activity that could harm MoJ's reputation or compromise the security of data or IT systems.
 
-- Data Owners are ultimately responsible for who has access to data while in the platform.
-- Data Owners **must** actively manage delegated access for team members (popularly - Joiners, Movers, Leavers)
-- Self-managed permissions through [data access groups](https://moj-analytical-services.github.io/platform_user_guidance/user-administration.html#teams-and-data-access-groups) must be used to provision and remove access as suitable - which is solely defined by Data Owners.
+__App admins and data source admins will:__
 
-## Data Minimisation
+* ensure that app and data source users have the correct read/write permissions;
+* ensure that app and data source users only have access to the minimum data required for them to perform their job; and
+* regularly review access permissions for app and data source users, including when users join or leave MoJ, or move within MoJ.
 
-- Consider [minimising data](https://moj-analytical-services.github.io/platform_user_guidance/annexes.html#data-minimisation) through pseudonymisation and anonymisation techniques.
+## GitHub
 
-## Guidance on the use of Github
+GitHub can be used to store code with a security marking of OFFICIAL only. Where possible, you should store data outside of GitHub even when it is not covered by the below restrictions.
```

Comment:
We don't mark code, so people will be confused by this (yes it is all OFFICIAL).

People _must not_ store data in Github, so "where possible" is too relaxed.

They do need permission before opensourcing, so that needs to stay in as well.

I'm tempted to suggest sticking with the original content even if you finesse the phrasing within.

Reply:
A couple of questions/comments:
1. Can all source code be stored in GitHub even if it contains sensitive information i.e. `prisons_to_close <- c("prison_1", "prison_2", ...)`?
2. In the current version of the guidance, it says that users can store small, non-sensitive data sets in GitHub. Do you no longer want that to be the case?

Reply:
> 1. Can all source code be stored in GitHub even if it contains sensitive information i.e. `prisons_to_close <- c("prison_1", "prison_2", ...)`?

This depends on the context of the code - https://ministryofjustice.github.io/technical-guidance/standards/storing-source-code/#github

This is why it requires author and line manager judgement on whether to open source or not.

> 2. In the current version of the guidance, it says that users can store small, non-sensitive data sets in GitHub. Do you no longer want that to be the case?

Hmmm. I'm not sure. One for Sam. I think we should remove that = no data in github fullstop.

Reply:
Presumably this is because any data in github is at risk, and having a clear "no data in github" policy minimises this risk?

I think it makes for much easier ways of working, particularly early on in a project when working with small, test data sets, if they can be stored in github. Particularly with relatively new platform users, since otherwise they have to learn both github and our methodology for accessing S3 before they can do anything.

But if you feel strongly about it from a security perspective, I'm happy to go with your recommendation. I can see there is a risk of falling back to people's judgement as to what constitutes a 'small, non-sensitive data set'.

Reply:
Meet half-way? <1000 rows (aggregate) of non-sensitive non-personal data can be stored in a github.com *private* repo?

It is a security concern, but its also an information management one - its very easy for data to be sat somewhere no one ever knows about forever etc.

Reply:
Further to this, I've just come from a meeting where we were talking about using github and our existing CI/CD processes to start recording lookup tables for data. As part of this, the tables would have to be uploaded to github in the first instance, before concourse would check them against some metadata and assess their data quality, and then push them to S3 and turn them into a database.

The question is whether we are happy with pushing lookup tables about our data to github? Obviously private repos only, and there's a question as to how sensitive lookup tables are or could be.

Reply:
Meeting halfway seems sensible. I've fleshed this section out a bit in f2ab04b. @samtazzyman it sounds like this might be something to discuss as a separate issue once the idea is firmed up a bit.

Reply:
Agreed, @calumabarnett @samtazzyman the lookup tables bit in github.com seems a bit different and needs some consideration